client/landing/stepper/declarative-flow/site-assembler-flow.ts

```diff
@@ -0,0 +1,68 @@
+import { useFlowProgress, ASSEMBLER_FLOW } from '@automattic/onboarding';
+import { useDispatch } from '@wordpress/data';
+import { useEffect } from 'react';
+import { useQuery } from '../hooks/use-query';
+import { useSiteSlug } from '../hooks/use-site-slug';
+import { ONBOARD_STORE } from '../stores';
+import { recordSubmitStep } from './internals/analytics/record-submit-step';
+import PatternAssembler, {
+	BLANK_CANVAS_DESIGN,
+} from './internals/steps-repository/pattern-assembler';
+import Processing from './internals/steps-repository/processing-step';
+import { Flow, ProvidedDependencies } from './internals/types';
+import type { Design } from '@automattic/design-picker/src/types';
+
+const siteAssemblerFlow: Flow = {
+	name: ASSEMBLER_FLOW,
+	useSteps() {
+		const { setSelectedDesign } = useDispatch( ONBOARD_STORE );
+		const selectedTheme = useQuery().get( 'theme' );
+
+		useEffect( () => {
+			if ( selectedTheme === 'blank-canvas-3' && this.name === ASSEMBLER_FLOW ) {
+				// User has selected blank-canvas-3 theme from theme showcase and enter assembler flow
+				setSelectedDesign( BLANK_CANVAS_DESIGN as Design );
+			}
+		}, [] );
+
+		return [
+			{ slug: 'patternAssembler', component: PatternAssembler },
+			{ slug: 'processing', component: Processing },
+		];
+	},
+
+	useStepNavigation( _currentStep, navigate ) {
+		const flowName = this.name;
+		const { setStepProgress, setPendingAction } = useDispatch( ONBOARD_STORE );
+		const flowProgress = useFlowProgress( { stepName: _currentStep, flowName } );
+		setStepProgress( flowProgress );
+		const siteSlug = useSiteSlug();
+
+		const exitFlow = ( to: string ) => {
+			setPendingAction( () => {
+				return new Promise( () => {
+					window.location.assign( to );
+				} );
+			} );
+
+			return navigate( 'processing' );
+		};
+
+		const submit = ( providedDependencies: ProvidedDependencies = {} ) => {
+			recordSubmitStep( providedDependencies, '', flowName, _currentStep );
```

Comment:
Would it be better to introduce new intent for the site assembler flow? Or is the flow name enough for tracking cc @autumnfjeld

Reply:
I think the flow name should be enough for now cc: @autumnfjeld

Reply:
@arthur791004 I think we can leave this as it is and address it on metrics issue, how do you think?

Reply:
Sure!

Reply:
Great!

Reply:
We have discussed the topic of intent in a few different places:
* p1671807817094079/1671770959.129519-slack-CRWCHQGUB
* pbxlJb-3bS-p2#comment-2201
* p1673498055131429-slack-C048CUFRGFQ


The problem with using `intent=assembler` is that we can't keep track of the build and write intent if we overwrite it with `assembler`. And the write intent is used in My Home to conditionally show content.   Thus, we have some constraints we must respect.  

Can you point to all the different `flowName` options we have?  To give me a sense of where else flow is used.

Is it this same flow. And [these](https://github.com/Automattic/wp-calypso/blob/a3402293c7319398fe5d1a60d0f183e17e6e55f6/client/landing/stepper/declarative-flow/registered-flows.ts) are all the flow possibilities?

<img width="335" alt="image" src="https://user-images.githubusercontent.com/5665959/215621943-b26bc94b-9421-415e-a55c-03c856f7034c.png">


I'm worried we are using flow in two different ways:
* stepper flow definition
* singup flow definition (e.g. when `flow=onboarding`)
* maybe these are all compatible? 
* see [FigJam](58MP46f44oGKl5Bin5bSp1-fi-0%3A1) (which I didn't yet update! oops!)
* 
<img width="1520" alt="image" src="https://user-images.githubusercontent.com/5665959/215622366-33fb7e2c-9d41-4b28-8ae1-814e151337a6.png">

Reply:
I started a[ new metrics map](58MP46f44oGKl5Bin5bSp1-fi-0%3A1) for the logged out PA flow.  Let's fill in all the items correctly:

![image](https://user-images.githubusercontent.com/5665959/215624736-f9ec7930-eb68-4360-8a1b-513044bda668.png)